src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -173,48 +173,92 @@ protected override Conversion GetInterpolatedStringConversion(BoundExpression so
                     break;
             }
 
-            Debug.Assert(collectionTypeKind == CollectionExpressionTypeKind.CollectionInitializer || elementType is { });
-
-            if (collectionTypeKind == CollectionExpressionTypeKind.CollectionInitializer)
+            var elements = node.Elements;
+            if (collectionTypeKind == CollectionExpressionTypeKind.ImplementsIEnumerable)
+            {
+                return Conversion.CreateCollectionExpressionConversion(collectionTypeKind, elementType: null, default);
+            }
+            else if (collectionTypeKind == CollectionExpressionTypeKind.ImplementsIEnumerableT)
             {
-                var implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, targetType) { WasCompilerGenerated = true };
-                var diagnostics = BindingDiagnosticBag.Discarded;
-                var collectionInitializerAddMethodBinder = _binder.WithAdditionalFlags(BinderFlags.CollectionInitializerAddMethod);
-                foreach (var element in node.Elements)
+                ImmutableArray<NamedTypeSymbol> allInterfaces;
+                switch (targetType.TypeKind)
+                {
+                    case TypeKind.Class:
+                    case TypeKind.Struct:
+                        allInterfaces = targetType.AllInterfacesNoUseSiteDiagnostics;
+                        break;
+                    case TypeKind.TypeParameter:
+                        allInterfaces = ((TypeParameterSymbol)targetType).AllEffectiveInterfacesNoUseSiteDiagnostics;
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(targetType.TypeKind);
+                }
+
+                var ienumerableType = this.Compilation.GetSpecialType(SpecialType.System_Collections_Generic_IEnumerable_T);
+                bool isCompatible = false;
+                foreach (var @interface in allInterfaces)
                 {
-                    _ = _binder.BindCollectionExpressionElementAddMethod(
-                        element,
-                        collectionInitializerAddMethodBinder,
-                        implicitReceiver,
-                        diagnostics,
-                        out bool hasErrors);
-                    if (hasErrors)
+                    if (isCompatibleIEnumerableT(@interface, ienumerableType, elements, ref useSiteInfo))
                     {
-                        return Conversion.NoConversion;
+                        isCompatible = true;
```

Comment:
`break;`?

Reply:
No, see Chuck's comment above. If we break, we don't report the remaining use-site diagnostics.

Reply:
I see. Please leave a comment to clarify, because it currently looks like you missed something :slightly_smiling_face:.